### PR TITLE
Cover asset_caching by tests

### PR DESCRIPTION
**Pull request by okurz**
* Add simple start-up smoke test for worker caching

### CODE REVIEW

File: t/24-worker-engine.t [hunk 1]
```diff
@@ -103,5 +107,11 @@ subtest 'asset settings' => sub {
     is_deeply($got, $expected, 'Asset settings are correct (no UEFI or NUMDISKS)') or diag explain $got;
 };
 
+subtest 'caching' => sub {
+    is(OpenQA::Worker::Engines::isotovideo::cache_assets, undef, 'cache_assets has nothing to do without assets');
+    my %assets = (ISO => 'foo.iso',);
+    my $got    = OpenQA::Worker::Engines::isotovideo::cache_assets(undef, undef, \%assets, undef, undef);
```
Comment (Amrysliu) [inline t/24-worker-engine.t:113]: I have a question about the second undef. The $vars is undef when doing $vars->{}. Is it better to pass a defined values to $vars?

File: t/24-worker-engine.t [hunk 2]
```diff
@@ -18,10 +18,14 @@ use warnings;
...
 use FindBin;
 use lib ("$FindBin::Bin/lib", "$FindBin::Bin/../lib");
+use Test::Exception;
 use Test::Fatal;
 use Test::More;
+use Test::Output qw(stdout_like stderr_like);
 use Test::Warnings;
 use OpenQA::Worker;
+use Test::MockModule;
+use Test::MockObject;
```
Comment (Amrysliu) [inline t/24-worker-engine.t:28]: Do we need these modules?

File: t/24-worker-engine.t (hunk 1 above)
Comment (okurz) [reply t/24-worker-engine.t:113]: I guess it's ok this way for testing.

File: t/24-worker-engine.t (hunk 2 above)
Comment (okurz) [reply t/24-worker-engine.t:28]: Right. These are only needed in https://github.com/os-autoinst/openQA/pull/2503 in another commit. I will change this, thx.

**Pull request merged**